src/components/languageSwitch.tsx

```diff
@@ -0,0 +1,24 @@
+'use client';
+import { ChangeEvent } from 'react';
+import { useRouter, usePathname } from 'next-intl/client';
+import { useTransition } from 'react';
+import LOCALES from '@/constants/locales';
+import Select from './select';
+
+function LanguageSwitch() {
+  const [isPending, startTransition] = useTransition();
+
+  const router = useRouter();
+  const pathname = usePathname();
+
+  const handleSelect = ({ target }: ChangeEvent<HTMLSelectElement>) => {
+    const lang = target.value;
+    startTransition(() => {
+      router.replace(pathname, { locale: lang });
+    });
+  };
+
+  return <Select options={LOCALES} handleSelect={handleSelect} />;
```

Comment:
options={localesOptions}

Reply:
i trzeba je zrobić z key i value

Reply:
czym ma być w tym przypadku key?

Reply:
tym co będzie wysyłane jako wartość na backend np w przyszłości, a value możeby traktować jako label to co będzie wyświetlane jako opcja możemy też to zamienić i dla czytelności zrobić `value` i `label`

czyli wtedy np: [{value: "pl", label: "Polski"}], w naszym przypadku tym akurat specyficznym value i label mogą być takie same, ale dobrze sobie to już na tym etapie rozróżnić, bo w sytuacji gdy będziemy coś wysyłać na backend co niekoniecznie będzie się nadawać do wyświetlenia jako label, albo będzie wymagało by label był tłumaczony, takie rozróżnienie się dobrze sprawdzi